features/dd-sdk-android-rum/src/test/kotlin/com/datadog/android/rum/internal/metric/RumSessionEndedMetricTest.kt

```diff
@@ -0,0 +1,412 @@
+    ) {
+        // Given
+        val duration1 = TimeUnit.SECONDS.toNanos(15)
+        val duration2 = TimeUnit.SECONDS.toNanos(10)
+        val duration3 = TimeUnit.SECONDS.toNanos(5)
+        val date1 = TimeUnit.SECONDS.toMillis(20)
+        val date2 = TimeUnit.SECONDS.toMillis(30)
+        val date3 = TimeUnit.SECONDS.toMillis(40)
+        val sessionEvent1 = event1.copy(
+            session = event1.session.copy(id = fakeSessionId),
+            date = date1,
+            view = event1.view.copy(timeSpent = duration1)
+        )
+        val sessionEvent2 = event2.copy(
+            session = event2.session.copy(id = fakeSessionId),
+            date = date2,
+            view = event2.view.copy(timeSpent = duration2)
+        )
+        val sessionEvent3 = event3.copy(
+            session = event3.session.copy(id = fakeSessionId),
+            date = date3,
+            view = event3.view.copy(timeSpent = duration3)
+        )
+        val sessionEndedMetric = SessionEndedMetric(fakeSessionId, fakeStartReason)
+
+        // When
+        sessionEndedMetric.onViewTracked(sessionEvent1)
+        sessionEndedMetric.onViewTracked(sessionEvent2)
+        sessionEndedMetric.onViewTracked(sessionEvent3)
+
+        // Then
+        val attributes = sessionEndedMetric.toMetricAttributes()
+        val rse = attributes[SessionEndedMetric.RSE_KEY] as Map<*, *>
+        val duration = rse[SessionEndedMetric.DURATION_KEY] as Long
+        val expected = TimeUnit.MILLISECONDS.toNanos(date3 - date1) + duration3
+        assertEquals(expected, duration)
+    }
+
+    @Test
+    fun `M ignore W track different session id view event`(
+        @Forgery event1: ViewEvent,
+        @Forgery event2: ViewEvent
+    ) {
+        // Given
+        val duration1 = TimeUnit.SECONDS.toNanos(15)
+        val duration2 = TimeUnit.SECONDS.toNanos(10)
+        val date1 = TimeUnit.SECONDS.toMillis(20)
+        val date2 = TimeUnit.SECONDS.toMillis(30)
+
+        val currentSessionViewEvent = event1.copy(
+            session = event1.session.copy(id = fakeSessionId),
+            date = date1,
+            view = event1.view.copy(timeSpent = duration1)
+        )
+        val differentSessionViewEvent = event2.copy(
+            date = date2,
+            view = event2.view.copy(timeSpent = duration2)
+        )
+        val sessionEndedMetric = SessionEndedMetric(fakeSessionId, fakeStartReason)
+
+        // When
+        sessionEndedMetric.onViewTracked(currentSessionViewEvent)
+        sessionEndedMetric.onViewTracked(differentSessionViewEvent)
+
+        // Then
+        val attributes = sessionEndedMetric.toMetricAttributes()
+        val rse = attributes[SessionEndedMetric.RSE_KEY] as Map<*, *>
+        val viewCounts = rse[SessionEndedMetric.VIEW_COUNTS_KEY] as Map<*, *>
+        assertEquals(1, viewCounts[SessionEndedMetric.VIEW_COUNTS_TOTAL_KEY])
+    }
+
+    @Test
+    fun `M view counts total correct W track views`(@Forgery events: List<ViewEvent>) {
```

Comment:
`M` stands for must, so the phrase after it should have a verb. Overall, other tests names in this file should be updated.

```suggestion
    fun `M calculate total view count W onViewTracked`(@Forgery events: List<ViewEvent>) {
```